### PR TITLE
Feat/7 login with GitHub oauth

### PR DESCRIPTION
**Pull request by DamienKusters**
Add Login and Registering with GitHub.

### CODE REVIEW

File: lpr.Logic/Services/JWTService.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+            JwtSecurityTokenHandler jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
+            try
+            {
+                ClaimsPrincipal tokenValid = jwtSecurityTokenHandler.ValidateToken(token, tokenValidationParameters, out SecurityToken validatedToken);
+                return true;
+            }
+            catch (Exception)
+            {
+                return false;
```
Comment (WouterPennings) [inline lpr.Logic/Services/JWTService.cs:74]: Dont you need to log/print all the exceptions in console?

File: lpr.WebAPI/Application.cs [hunk 2]
```diff
@@ -6,59 +6,115 @@
...
+          options.UseMySql(connectionString,
+                          new MariaDbServerVersion(new Version(10, 5, 9)));
+        });
+
+        builder.Services.AddScoped<IPackageService, PackageService>();
+        builder.Services.AddScoped<IPackageData, PackageData>();
+
+        using (var scope =
+                    builder.Services.BuildServiceProvider().CreateScope()) {
+          using (var context =
```
Comment (WouterPennings) [inline lpr.WebAPI/Application.cs:30]: Isnt there a new more eligant way to use "using"?

File: lpr.WebAPI/Middleware/JWTMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                    IEnumerable<Claim> claims = jwtService.GetTokenClaims(token);
+                    string githubId = claims.FirstOrDefault(e => e.Type.Equals(ClaimTypes.Authentication)).Value;
+
+                    context.Items["GitHubAccessToken"] = githubId;
+                    //context.Items["GitHubAccessToken"] = dbContext.Account.FirstOrDefault(x => x.GithubId == githubId);
+                }
+            }
+            catch
+            {
+                //ignore catch
```
Comment (WouterPennings) [inline lpr.WebAPI/Middleware/JWTMiddleware.cs:32]: No Log/print needed?

**Review by WouterPennings — COMMENTED** [review]
Looks good, I will let @NealGeilen decide

File: lpr.Logic/Services/JWTService.cs (hunk 1 above)
Comment (DamienKusters) [reply lpr.Logic/Services/JWTService.cs:74]: We don't have a 'official' method of logging errors yet.
So we might look into this next sprint.

File: lpr.WebAPI/Middleware/JWTMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            string token = context.Request.Headers["X-JWT-Token"];
+            try
+            {
+                if (jwtService.IsTokenValid(token))
+                {
+                    IEnumerable<Claim> claims = jwtService.GetTokenClaims(token);
+                    string githubId = claims.FirstOrDefault(e => e.Type.Equals(ClaimTypes.Authentication)).Value;
+
+                    context.Items["GitHubAccessToken"] = githubId;
```
Comment (nickonos) [inline lpr.WebAPI/Middleware/JWTMiddleware.cs:26]: This should account id instead of GitHub id

File: lpr.WebAPI/Application.cs (hunk 2 above)
Comment (DamienKusters) [reply lpr.WebAPI/Application.cs:30]: It might be,
The Application.cs file is prune to rapid changes.
So optimizing the file is often quickly discarded.

Comment (DamienKusters) [reply lpr.WebAPI/Application.cs:30]: But we'll look into this.

File: lpr.WebAPI/Attributes/AuthenticatedAttribute.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+using System;
+
+namespace lpr.WebAPI.Attributes
+{
+    [AttributeUsage(AttributeTargets.Method)]
+    public class AuthenticatedAttribute : Attribute, IAuthorizationFilter
+    {
+        public void OnAuthorization(AuthorizationFilterContext filterContext)
+        {
+            string user = (string)filterContext.HttpContext.Items["GitHubAccessToken"];
```
Comment (nickonos) [inline lpr.WebAPI/Attributes/AuthenticatedAttribute.cs:13]: This should be chaged to account id

**Review by nickonos — CHANGES_REQUESTED** [review]
See the comments below

File: lpr.WebAPI/Middleware/JWTMiddleware.cs (hunk 4 above)
Comment (DamienKusters) [reply lpr.WebAPI/Middleware/JWTMiddleware.cs:26]: Correct this is a bug.

**Review by NealGeilen — APPROVED** [review]

**Review by WouterPennings — APPROVED** [review]

**Review by nickonos — APPROVED** [review]

**Pull request merged**